sql/core/src/main/scala/org/apache/spark/sql/util/QueryExecutionListener.scala

```diff
@@ -124,13 +131,27 @@ class ExecutionListenerManager private[sql](session: SparkSession, loadExtension
   }
 }
 
-private[sql] class ExecutionListenerBus(session: SparkSession)
```

Comment:
IIUC, `ExecutionListenerBus` is created per SparkSession intentionally. With your change, it seems to break the assumption of "Only catch SQL execution with a name, and triggered by the same spark session that this listener manager belongs."

Reply:
We don't break the assumption with this change. SQL executions will catch by the launched session listener manager because we only send the `SparkListenerSQLExecutionStart` event to the launched session listener manager (see the change in the file `SQLExecution.scala`)  and process the `SparkListenerSQLExecutionEnd` event with the same listener manager.